### ISSUE TITLE
[F.Rq.] Accept ZX Spectrum programs in the `.z80` format

### ISSUE THREAD

**Issue by YoshiRulz** [TASEmulators/BizHawk#4016]
https://worldofspectrum.org/faq/reference/z80format.htm
> arguably the most widely supported by emulators across all platforms

This doesn't appear to be supported by the core, but I'm not certain.

**Comment by Asnivor** [comment]
The consensus when ZXS was developed was that (for TASing at least) consoles and computers should implement loading and timing of commercially available media of the original machines. In the case of the Spectrum this was either disk or tape. There were peripherals that would allow the saving and loading of snapshots, but games were never released in this format back in the day.

This is ultimately why the various memory snapshot formats (.z80, .sna etc) have not been implemented (and why I haven't implemented enhanced loading speed for tape media).

The snapshot formats are all fairly trivial to implement, but we would probably need a way to make them casual-only. Unless the general thinking has changed with regards to TASing.

**Comment by YoshiRulz** [comment]
[TASVideos' submission rules](https://tasvideos.org/MovieRules) don't explicitly cover the ZX Spectrum or tape vs. disk, but I imagine that for the purposes of obsoletion, if a difference in tape loading time did happen it would be excluded from consideration, like loading times or a difference due to text language.

**Comment by Asnivor** [comment]
I think it's more likely to be a problem that it's not an official 'rom'. *.z80 is really just a memory snapshot with some CPU register info (so nowhere near what the data on the original media would look like). It's a format that isn't playable on original hardware, and I think this is where it will be outright banned for TAS submissions.

Perhaps there are some edge cases where modern games were released in a snapshot format, but they almost always have a *.tap, *.tzx or *.dsk available as well. If not, then perhaps that is allowed on a case-by-case basis.

Maybe some flag is needed within the core that causes a user to be informed that the loaded snapshot is not TAS-appropriate - perhaps when TAStudio is opened or whatever?

**Comment by YoshiRulz** [comment]
Ah, you meant only third-party peripherals can dump in that format. In that case I think you're right.

We have a mechanism for blocking recording (which stands to be improved), and we also have the profile system. Let's leave the bikeshedding over that for after the feature is implemented.

**Comment by vadosnaprimer** [comment]
> I think it's more likely to be a problem that it's not an official 'rom'. *.z80 is really just a memory snapshot with some CPU register info (so nowhere near what the data on the original media would look like). It's a format that isn't playable on original hardware, and I think this is where it will be outright banned for TAS submissions.

Yeah sounds illegitimate. We're emulating a device. If said device can't even run those things, we can't call them ZX Spectrum either.

**Comment by Asnivor** [comment]
I understand the potential need for it. It is one of the defacto standard saving and loading formats in just about every other spectrum emulator. We probably should have it for casual play (along with *.szx - the newer snapshot format).
